### PR TITLE
Add Chat Types

### PR DESCRIPTION
**Pull request by Gaurav-Malviya**
I went through the discord channel and saw some functionality which it has. Based on that  I have added two parameter on top of the recommended parameter.  They are: 1)muteChannel--> which takes enum (duration of mute) as an input 2)inviteLink--> an invitation link for others to join

### CODE REVIEW

File: channel/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import {User} from "../user";
```
Comment (AntoineKM) [inline channel/index.d.ts:1]: your file has to be ``index.d.ts``

**Review by AntoineKM — CHANGES_REQUESTED** [review]
you need to use eslint or use theses command:
- ``yarn lint``
- ``yarn lint:fix``

File: channel/index.d.ts (hunk 1 above)
Comment (Gaurav-Malviya) [reply channel/index.d.ts:1]: Isn't the file name index.d.ts and I will do the linting

**Comment by Gaurav-Malviya** [comment]
Now the code is clean... 
Please review it

**Pull request closed without merging**